transactions/src/main/java/org/jboss/as/txn/service/ArjunaRecoveryManagerService.java

```diff
@@ -26,20 +26,25 @@
 import java.util.Collections;
 import java.util.List;
 
+import com.arjuna.ats.arjuna.recovery.RecoveryManager;
 import com.arjuna.ats.internal.jta.recovery.jts.JCAServerTransactionRecoveryModule;
 import org.jboss.as.network.ManagedBinding;
 import org.jboss.as.network.SocketBinding;
 import org.jboss.as.network.SocketBindingManager;
 import org.jboss.as.server.suspend.SuspendController;
 import org.jboss.as.txn.logging.TransactionLogger;
 import org.jboss.as.txn.suspend.RecoverySuspendController;
+import org.jboss.jbossts.xts.initialisation.RecoveryInitialisation;
+import org.jboss.jbossts.xts.recovery.participant.ba.XTSBARecoveryManager;
```

Comment:
The reason is that the compensations is capable to cooperate with XTS. The `@Compensatable` annotation is used for handling local saga or cooperate with WF business activity. When compensating recovery modules should be started you don't know if the failed `@Compensatable` is for the case of the local or remote. If it's about to finish the remote business activity, you need the XTS code to do so.